src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -137,22 +273,41 @@ protected internal override void Write(int pinNumber, PinValue value)
 
         protected override void Dispose(bool disposing)
         {
-            foreach (int pin in _pinNumberToSafeLineHandle.Keys)
+            _disposing = disposing;
+
+            if (_pinNumberToEventHandler != null)
             {
-                if (_pinNumberToSafeLineHandle.TryGetValue(pin, out SafeLineHandle pinHandle))
+                foreach (var kv in _pinNumberToEventHandler)
                 {
-                    Interop.ReleaseGpiodLine(pinHandle);
+                    int pin = kv.Key;
+                    LibGpiodDriverEventHandler eventHandler = kv.Value;
+                    eventHandler.Dispose();
+                    if (_pinNumberToSafeLineHandle.TryGetValue(pin, out SafeLineHandle pinHandle))
+                    {
+                        _pinNumberToSafeLineHandle.Remove(pin);
+                    }
                 }
 
-                _pinNumberToSafeLineHandle.Remove(pin);
+                _pinNumberToEventHandler = null;
             }
 
-            if (_chip != null)
+            
+            if (_pinNumberToSafeLineHandle != null)
             {
-                _chip.Dispose();
-                _chip = null;
+                foreach (int pin in _pinNumberToSafeLineHandle.Keys)
+                {
+                    if (_pinNumberToSafeLineHandle.TryGetValue(pin, out SafeLineHandle pinHandle))
```

Comment:
When would this return false?

Reply:
when pin value was not present for that pin

Reply:
@buyaa-n shouldn't we never get in this state and validate the input parameters as part of accepting pin values prior to gettting all the way to Dispose?